dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -477,102 +410,128 @@ public List<TrackedEntity> getTrackedEntities(
             queryParams,
             operationParams.getOrgUnitMode());
 
-    mapRelationshipItems(
+    setRelationshipItems(
         trackedEntities,
         operationParams.getTrackedEntityParams(),
         operationParams.isIncludeDeleted());
+    for (TrackedEntity trackedEntity : trackedEntities) {
+      getTrackedEntityAttributeValues(trackedEntity, queryParams.getProgram());
+    }
 
     addSearchAudit(trackedEntities);
 
     return ids.withItems(trackedEntities);
   }
 
-  private List<Long> getTrackedEntityIds(TrackedEntityQueryParams params) {
-    return trackedEntityStore.getTrackedEntityIds(params);
-  }
-
-  private Page<Long> getTrackedEntityIds(TrackedEntityQueryParams params, PageParams pageParams) {
-    return trackedEntityStore.getTrackedEntityIds(params, pageParams);
-  }
-
   /**
    * We need to return the full models for relationship items (i.e. trackedEntity, enrollment and
    * event) in our API. The aggregate stores currently do not support that, so we need to fetch the
    * entities individually.
    */
-  private void mapRelationshipItems(
+  private void setRelationshipItems(
       List<TrackedEntity> trackedEntities, TrackedEntityParams params, boolean includeDeleted)
       throws NotFoundException {
+    for (TrackedEntity trackedEntity : trackedEntities) {
+      setRelationshipItems(trackedEntity, trackedEntity, params, includeDeleted);
+    }
+  }
+
+  private void setRelationshipItems(
+      TrackedEntity targetTrackedEntity,
+      TrackedEntity sourceTrackedEntity,
```

Comment:
I am confused by this target and source `trackedEntity` here.
When `setRelationshipItems` is called by `getTrackedEntities` method, source and target is the same `trackedEntity` and the nested for loops seem not necessary.
When `setRelationshipItems` is called by `getTrackedEntity`, source is the `trackedEntity`to be mapped and the target is the `trackedEntity` coming from the DB. At this point source `enrollments` and `events` are not set yet, so no relationship item will be set for enrollments and events even if the params are set to include relationhips.